include/aikido/robot/Hand.hpp

```diff
@@ -0,0 +1,109 @@
+#ifndef AIKIDO_ROBOT_HAND_HPP_
+#define AIKIDO_ROBOT_HAND_HPP_
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <dart/dart.hpp>
+#include "aikido/robot/GrabInfo.hpp"
+#include "aikido/io/yaml.hpp"
+#include "aikido/control/PositionCommandExecutor.hpp"
+
+namespace aikido {
+namespace robot {
+
+/// A base class for a hand used by manipulators
+class Hand
+{
+public:
+
+  /// \param newName New name for this robot
+  virtual std::unique_ptr<Hand> clone(const std::string& newName) override;
+
+  /// Get the end-effector body node.
+  /// \return DART body node of end-effector
+  dart::dynamics::BodyNode* getBodyNode() const;
+
+  /// Get the hand skeleton.
+  /// \return DART Branch rooted at palm of a BarrettHand
+  dart::dynamics::BranchPtr getHand();
+
+  /// Get the hand's offset from a TSR of a specific object type (as registered
+  /// in \c tsrEndEffectorTransformsUri). Each object's TSR.mTw_e must be
+  /// right-multiplied with this.
+  /// \param[in] _objectType Type of the object (e.g. "cylinder")
+  /// \return hand's transform if it exists, boost::none if not
+  boost::optional<Eigen::Isometry3d> getEndEffectorTransform(
+      const std::string& _objectType) const;
```

Comment:
Nit: `objectType` Let's not to use a leading underscore for argument names.

Reply:
@jslee02 does your comment above about const-correctness apply here as well? I am just curious, don't know much about it at this point but I'd assume the same problems mentioned there would occur?